### PR TITLE
[example] fix save_load bug for dreambooth

### PR DESCRIPTION
**Pull request by ht-zhou**

### CODE REVIEW

File: examples/images/dreambooth/colossalai.sh [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
   --train_batch_size=1 \
   --gradient_accumulation_steps=1 \
   --learning_rate=5e-6 \
-  --instance_prompt=INSTANCE_PROMPT \
   --lr_scheduler="constant" \
   --lr_warmup_steps=0 \
-  --max_train_steps=400 \
-  --placement="cpu"
+  --num_class_images=200 \
+  --placement="cuda" \
```
Comment (feifeibear) [inline examples/images/dreambooth/colossalai.sh:22]: Can it works use "cpu"?

File: examples/images/dreambooth/train_dreambooth_colossalai.py [hunk 2]
```diff
@@ -31,7 +32,7 @@
 from colossalai.nn.parallel.utils import convert_to_torch_module
 from colossalai.tensor import ColoTensor, ProcessGroup
 from colossalai.utils import get_current_device
-from colossalai.utils.model.colo_init_context import ColoInitContext
+from colossalai.utils.model.colo_init_context import ColoInitContext, post_process_colo_init_ctx
```
Comment (feifeibear) [inline examples/images/dreambooth/train_dreambooth_colossalai.py:35]: remove post_process_colo_init_ctx, not used

File: examples/images/dreambooth/colossalai.sh (hunk 1 above)
Comment (ht-zhou) [reply examples/images/dreambooth/colossalai.sh:22]: let me have a try

File: examples/images/dreambooth/train_dreambooth_colossalai.py [hunk 3]
```diff
@@ -653,7 +656,7 @@ def collate_fn(examples):
 
             optimizer.step()
             lr_scheduler.step()
-
+            print(torch.cuda.max_memory_allocated()/2**20)
```
Comment (feifeibear) [inline examples/images/dreambooth/train_dreambooth_colossalai.py:659]: use logger.info( XX, ranks = [0])

File: examples/images/dreambooth/colossalai.sh (hunk 1 above)
Comment (ht-zhou) [reply examples/images/dreambooth/colossalai.sh:22]: It also works

**Review by feifeibear — APPROVED** [review]

**Pull request merged**